File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/Java/src/Model/Polygons/Pentagon.java
package Model.Polygons;

import Model.Line;
import Model.Point;

import java.util.List;

public class Pentagon extends Polygon{
    public Pentagon(final List<Point> vertices) throws Exception {
        super(vertices);
    }

    /**
     * Returns whether a list of points can make a pentagon or not
     */
    public static boolean canBeCreatedWith(final Polygon polygon) {
        return polygon.getVertices().size() == 5;
    }

    @Override
    public double getApothemConstant() throws Exception {
        return isRegular() ? 0.688 : super.getApothemConstant();
    }

    @Override
    public String toString() {
        return "Pentagon";
    }
}

File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/C++/Model/Polygons/Pentagon.cpp
#include "Pentagon.h"

Pentagon::Pentagon(const list<Point> &vertices) : Polygon(vertices) {}

bool Pentagon::canBeCreatedWith(const Polygon &polygon) {
    return polygon.getVertices().size() == 5;
}

double Pentagon::getApothemConstant() const {
    return isRegular() ? 0.688 : Polygon::getApothemConstant();
}

string Pentagon::getName() const {
    return "Pentagon";
}